plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -622,18 +622,21 @@ export class CatalogBuilder {
       enableRelationsCompatibility,
     });
 
-    if (config.getOptionalString('catalog.orphanProviderStrategy') !== 'keep') {
-      await evictEntitiesFromOrphanedProviders({
-        db: providerDatabase,
-        providers: entityProviders,
-        logger,
-      });
-    }
     await connectEntityProviders(providerDatabase, entityProviders);
```

Comment:
Hold up, should it be the case for connecting entity providers too?

Reply:
Yeah see the PR comment. I absolutely think that connecting providers should be optional in SOME way. It'd be breaking if people for example use providers to react to webhook events through the connection but don't want to also run processing on the same node. So I left it out for now, and maybe we should make it a more explicit config option for fine grained control?

Reply:
@drodil @Rugvip also connecting the dots to https://github.com/backstage/backstage/pull/30969

Maybe we could have something like

```yaml
catalog:
  processors: false
```

and

```yaml
catalog:
  processors:
    - name: MyProcessor
      enabled: false
```

or something along those lines - and the same for providers - to enable ordering and enablement etc one by one or as a whole

Reply:
Just small note that most of the providers use config where the provider name is the key of the config object, for example:

```ts
 catalog?: {
    providers?: {
      bitbucketServer?: {}
    }
}
```

Reply:
Yep. Started the work a bit with #31034 but it's still missing some of the features mentioned here. But it's easier to start small :)